src/Google.Pubsub.V1/SimpleTopic.cs

```diff
@@ -0,0 +1,113 @@
+        /// <returns>The message ID of the published message</returns>
+        public string Publish(ByteString data, IDictionary<string, string> attributes = null)
+        {
+            GaxPreconditions.CheckNotNull(data, nameof(data));
+            var message = new PubsubMessage { Data = data };
+            if (attributes != null)
+            {
+                message.Attributes.Add(attributes);
+            }
+            var response = Parent.Publisher.Publish(ResourceName, new[] { message });
+            return response.MessageIds[0];
+        }
+
+        /// <summary>
+        /// Asynchronously publishes a given Protobuf message to the topic, with optional attributes.
+        /// </summary>
+        /// <param name="message">The message to publish. Must not be null.</param>
+        /// <param name="attributes">The key/value dictionary of attributes to attach to the message.
+        /// May be null, in which case no attributes are added.</param>
+        /// <returns>A task representing the asynchronous operation, with a result returning the
+        /// message ID of the published message.</returns>
+        public Task<string> PublishAsync(IMessage message, IDictionary<string, string> attributes = null) =>
+            PublishAsync(GaxPreconditions.CheckNotNull(message, nameof(message)).ToByteString(), attributes);
+
+        /// <summary>
+        /// Asynchronously publishes a given text message to the topic by encoding it as UTF-8, with optional attributes.
+        /// </summary>
+        /// <param name="messageText">The text of the message to publish. Must not be null.</param>
+        /// <param name="attributes">The key/value dictionary of attributes to attach to the message.
+        /// May be null, in which case no attributes are added.</param>
+        /// <returns>A task representing the asynchronous operation, with a result returning the
+        /// message ID of the published message.</returns>
+        public Task<string> PublishAsync(string messageText, IDictionary<string, string> attributes = null) =>
+            PublishAsync(ByteString.CopyFromUtf8(GaxPreconditions.CheckNotNull(messageText, nameof(messageText))), attributes);
+
+        /// <summary>
+        /// Asynchronously publishes a message containing the given <see cref="ByteString"/> data, with optional attributes.
+        /// </summary>
+        /// <param name="data">The bytestring containing the data to publish in the message. Must not be null.</param>
+        /// <param name="attributes">The key/value dictionary of attributes to attach to the message.
+        /// May be null, in which case no attributes are added.</param>
+        /// <returns>A task representing the asynchronous operation, with a result returning the
+        /// message ID of the published message.</returns>
+        public async Task<string> PublishAsync(ByteString data, IDictionary<string, string> attributes = null)
```

Comment:
Cancellation?
Or are you missing it out because this is meant to be Simple?
(and in other async methods in this and other classes)